File: src/main/php/Easy/PalindromeNumber.php
<?php

declare(strict_types=1);

namespace Src\Easy;

class PalindromeNumber
{
    public function checkPalindrome(int $number): bool
    {
        if ($number < 0) {
            return false;
        }
        $value = strval($number);
        $right = strlen($value) - 1;
        for ($left = 0; $left < strlen($value) / 2; $left++) {
            if ($value[$left] != $value[$right]) {
                return false;
            }
            $right -= 1;
        }

        return true;
    }
}

File: src/easy/PalindromeNumber.swift
func palindromeNumber(_ x: Int) -> Bool {
  if x < 0 || (x % 10 == 0 && x != 0) {
    return false
  }
  let number = String(x)
  var left = number.startIndex
  var right = number.index(before: number.endIndex)
  while left < right {
    if number[left] != number[right] {
      return false
    }
    left = number.index(after: left)
    right = number.index(before: right)
  }
  return true
}
